pipefuser/utils.py

```diff
@@ -187,16 +191,38 @@ def __init__(self, distri_config: DistriConfig):
         self.send_req = None
         self.recv_req = None
         self.recv_queue = []
-        self.next_rank = (distri_config.rank + 1) % distri_config.world_size
-        self.prev_rank = (
-            distri_config.rank - 1 + distri_config.world_size
-        ) % distri_config.world_size
         self.dtype = None
-        if distri_config.world_size == 2:
-            self.groups = [torch.distributed.new_group(), torch.distributed.new_group()]
+
+
+        batch_world_size = dist.get_world_size()
+        self.batch_group = dist.new_group()
+
+        if distri_config.batch_group is not None and distri_config.split_group is not None:
```

Comment:
1. Use "split batch" by setting the dp degree and pipefusion degree
2. name the group for pipefusion as pipefusion_group. (split_group->pipefusion_group)
3. name the group for batch as dp_group. (batch_group -> dp_group)

Reply:
It feels like the DistriConfig is getting confusing as more and more options are added, is it possible to consider separating the configs for different parallel modes into different classes afterwards to make it easier to handle?

Reply:
OK, this PR, we can use the split batch flag. We can change the config in the future.